poker.js

```diff
@@ -0,0 +1,442 @@
+const _ = require("lodash");
+
+const listOfCards = {
+  2: 2,
+  3: 3,
+  4: 4,
+  5: 5,
+  6: 6,
+  7: 7,
+  8: 8,
+  9: 9,
+  T: 10,
+  J: 11,
+  Q: 12,
+  K: 13,
+  A: 14,
+};
+
+const combination = {
+  highCard: 100,
+  pair: 200,
+  twoPairs: 400,
+  threeOfAKind: 600,
+  straight: 800,
+  flush: 1000,
+  fullHouse: 1200,
+  fourOfAKind: 1400,
+  straightFlush: 1600,
+  royalFlush: 2000,
+};
+
+const processHands = (inputsArr) => {
+  /**
+     * user input cards, processHands will go through each card value,
+     * split them by space, separate them by assigning 1st element to cardValue and second element to suitValue
+     * then we split the cards by 5, 1st half belongs to player1 and the rest are for player2
+     * will look like this for each player [
+    { value: '2', suit: 'S' },
+    { value: 'K', suit: 'D' },
+    { value: 'T', suit: 'H' },
+    { value: '9', suit: 'H' },
+    { value: '8', suit: 'H' }
+  ]
+     */
+  return inputsArr.map((arr) => {
+    const splitArr = arr.split(" ").map((element) => {
+      return {
+        //split the user input (e.g. 9H) and set first string to be "value" and second string to be the "suit"
+        value: element.substring(0, 1),
+        suit: element.substring(1, 2),
+      };
+    });
+    //there's always 10 cards in each data input. Here we are dividing the data to 2 groups,
+    //first half belongs to player 1 and second half belongs to player 2
+    const player1Cards = splitArr.slice(0, 5);
+    const player2Cards = splitArr.slice(5, 10);
+
+    return { player1Cards, player2Cards };
+  });
+};
+
+const player1ScoreCard = [];
+let player1Score;
+const player2ScoreCard = [];
+let player2Score;
+
+const calculatePlayer1Score = (hand) => {
```

Comment:
calculatePlayer1Score, and calculatePlayer2Score are the same function, pass the player info into the function and remove the duplication

Reply:
done....